subprojects/parseq/build.gradle

```diff
@@ -35,6 +35,7 @@ configurations {
 }
 
 dependencies {
+  compile group: 'com.google.code.findbugs', name: 'jsr305', version: '3.0.2'
```

Comment:
why do we need to add this findbugs dependency? As we discussed earlier, I hope that this annotation should not depend on one particular static analysis tools.

Reply:
`javax.annotation` package is part of findbugs. This is just lib import, we don't need to run findbugs to use these annotations but if you're not willing to import lib as well then this change can't be made.
In my opinion importing this lib should not have any impact and it's a very standard practice.
let me know what you think.

Reply:
@mchen07 bump

Reply:
Sorry for the delay. I am still struggling whether we should introduce such @Nullable annotation where different IDE may have different support (for example, IntelliJ has its own @Nullable support, which is different from `javax.annotation`, see https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html), or instead we should use java Optional to indicate this, where we don't depend on the implementation of static code analysis tool.

Reply:
Hey @mchen07 Intellij Documentation doesn't mentions it (probably old) but IntelliJ support it. You can go to Intellij `Preferences -> Editor -> Inspections -> Probable bugs -> @NotNull/@Nullable problems -> Configure annotations`
There you can see `javax.annotation.Nullable` and `Notnull` too

Reply:
Thanks for the tip @hiteshsharma. Have you seen this: https://github.com/spotbugs/spotbugs/pull/180? Recently spotbugs is preferred over findBugs, will this have side effect on that trend?